packages/components/src/highlight-cards/weekly-highlight-cards.tsx

```diff
@@ -136,43 +216,10 @@ export default function WeeklyHighlightCards( {
 				) }
 
 				{ isHighlightsSettingsEnabled && (
-					<div className="highlight-cards-heading__settings">
-						<button
-							className="highlight-cards-heading__settings-action"
-							ref={ settingsActionRef }
-							onClick={ togglePopoverMenu }
-						>
-							<Icon className="gridicon" icon={ moreVertical } />
-						</button>
-						<Popover
-							className="tooltip highlight-card-popover"
-							isVisible={ isPopoverVisible }
-							position="bottom left"
-							context={ settingsActionRef.current }
-							focusOnShow={ false }
-						>
-							<button
-								onClick={ () => {
-									onTogglePeriod( PAST_SEVEN_DAYS );
-								} }
-							>
-								{ translate( '7-day highlights' ) }
-								{ currentPeriod === PAST_SEVEN_DAYS && (
-									<Icon className="gridicon" icon={ check } />
-								) }
-							</button>
-							<button
-								onClick={ () => {
-									onTogglePeriod( PAST_THIRTY_DAYS );
-								} }
-							>
-								{ translate( '30-day highlights' ) }
-								{ currentPeriod === PAST_THIRTY_DAYS && (
-									<Icon className="gridicon" icon={ check } />
-								) }
-							</button>
-						</Popover>
-					</div>
+					<HighlightCardsSettings
+						currentPeriod={ currentPeriod }
+						onTogglePeriod={ onTogglePeriod }
```

Comment:
What do you think about persisting the selected highlight type (7-day vs. 30-day) on the backend?

Reply:
Determining the data period via parameters to the API point could reduce the waiting time for fetching all data, let's iterate it in the near future. 🙂

Reply:
Setting persisting API is ready for some eyes: D112693-code